src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -0,0 +1,100 @@
+package gov.cabinetoffice.gap.adminbackend.services;
+
+import gov.cabinetoffice.gap.adminbackend.config.UserServiceConfig;
+import gov.cabinetoffice.gap.adminbackend.dtos.schemes.SchemeEditorsDTO;
+import gov.cabinetoffice.gap.adminbackend.dtos.user.UserEmailRequestDto;
+import gov.cabinetoffice.gap.adminbackend.dtos.user.UserEmailResponseDto;
+import gov.cabinetoffice.gap.adminbackend.entities.GrantAdmin;
+import gov.cabinetoffice.gap.adminbackend.entities.SchemeEntity;
+import gov.cabinetoffice.gap.adminbackend.enums.SchemeEditorRoleEnum;
+import gov.cabinetoffice.gap.adminbackend.repositories.SchemeRepository;
+import gov.cabinetoffice.gap.adminbackend.services.encryption.AwsEncryptionServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class SchemeEditorService {
+    private final RestTemplate restTemplate;
```

Comment:
I think going forwards we should use springs `WebClient` to make http requests. This is springs new way of making requests, and rest template has caused an issue before (I dont think it would cause the same issue here but still defo a better practice)